curvefs/src/metaserver/register.cpp

```diff
@@ -62,11 +62,10 @@ int Register::RegisterToMDS(MetaServerMetadata *metadata) {
     char hostname[128];
     int ret = gethostname(hostname, sizeof(hostname));
     if (ret == -1) {
-        LOG(ERROR) << "get hostname failed!";
-        return -1;
+        LOG(INFO) << "get hostname failed!";
+    } else {
+        req.set_hostname(hostname);
```

Comment:
You need change the hostName of MetaServerRegistRequest in topology.proto to optional.

Reply:
ok